extensions/amp-sidebar/1.0/sidebar.js

```diff
@@ -123,13 +148,32 @@ function SidebarWithRef(
     [open, close, toggle]
   );
 
+  useLayoutEffect(() => {
+    if (side) {
+      return;
+    }
+    const sidebarElement = sidebarRef.current;
+    if (!sidebarElement) {
+      return;
+    }
+    setSide(calculateSide(sideProp, sidebarElement.ownerDocument));
```

Comment:
Do a single calculation for `side` here.  Does some minor error checking.

If `sideProp` is not one of the Enum values, `left` or `right` it will use the defaulting logic.  Defaulting logic uses `document.dir` to determine `left` or `right` (defaults to `left` if unavailable).

Long story short, if user provides a mispelled input like `lfet`, it will default to `document.dir`'s value, which I think is preferred, than erroring out.